src/datasets.cpp

```diff
@@ -804,10 +804,17 @@ int SeedGenerator::new_seed()
      * Obtain a new non-negative integer seed
      * (returns -1 if SeedGenerator is in non-deterministic mode).
      */
+    // Obtain random number from hardware
+    std::random_device rd;
+    // Seed generator
+    std::mt19937 eng(rd());
```

Comment:
This shouldn't happen here, but in the `SeedGenerator` constructor. The `SeedGenerator` takes a meta-seed which it uses to seed a master-generator. If _that_ `meta_seed == -1`, then the master-generator in the constructor is seeded with `std::random_device rd;`. The constructor should create the random number generator and seed it with a master seed, then `new_seed()` pulls the next quasi-random (because it's seeded with the master-seed) seed from that generator and returns it.

Instead, `new_seed()` should pretty much only do:
```
if (this->meta_seed == -1){
  return rd();
}else{
  return this->distr(this->eng);
}
```

Reply:
Updated